src/resources/transactions/consolidatedTransactions.ts

```diff
@@ -0,0 +1,183 @@
+// Query Types
+
+export type ConsolidatedTransactionsArgs = {
+  address: string;
+  currency: NativeCurrencyKey;
+  chainIds: number[];
+};
+
+// ///////////////////////////////////////////////
+// Query Key
+
+export const consolidatedTransactionsQueryKey = ({
+  address,
+  currency,
+  chainIds,
+}: ConsolidatedTransactionsArgs) =>
+  createQueryKey(
+    'consolidatedTransactionss',
+    { address, currency, chainIds },
+    { persisterVersion: 1 }
+  );
+
+type ConsolidatedTransactionsQueryKey = ReturnType<
+  typeof consolidatedTransactionsQueryKey
+>;
+
+// ///////////////////////////////////////////////
+// Query Fetcher
+
+export async function fetchConsolidatedTransactions<
+  ConsolidatedTransactionsResult
+>(
+  { address, currency, chainIds }: ConsolidatedTransactionsArgs,
+  config: QueryConfig<
+    ConsolidatedTransactionsResult,
+    Error,
+    ConsolidatedTransactionsQueryKey
+  >
+) {
+  return await queryClient.fetchQuery(
+    consolidatedTransactionsQueryKey({
+      address,
+      currency,
+      chainIds,
+    }),
+    consolidatedTransactionsQueryFunction,
+    config
+  );
+}
+
+// ///////////////////////////////////////////////
+// Query Function
+
+type _QueryResult = {
+  cutoff?: number;
+  nextPage?: string;
+  transactions: RainbowTransaction[] | [];
+};
+
+export async function consolidatedTransactionsQueryFunction({
+  queryKey: [{ address, currency, chainIds }],
+  pageParam,
+}: QueryFunctionArgs<
+  typeof consolidatedTransactionsQueryKey
+>): Promise<_QueryResult> {
+  try {
+    const chainIdsString = chainIds.join(',');
+    const url = `https://addys.p.rainbow.me/v3/${chainIdsString}/${address}/transactions`;
+    const response = await rainbowFetch(url, {
+      method: 'get',
+      params: {
+        currency: currency.toLowerCase(),
+        ...(pageParam ? { pageCursor: pageParam } : {}),
+      },
+      timeout: CONSOLIDATED_TRANSACTIONS_TIMEOUT,
+      headers: {
+        Authorization: `Bearer ${ADDYS_API_KEY}`,
+      },
+    });
+
+    const consolidatedTransactions = await parseConsolidatedTransactions(
+      response?.data,
+      currency
+    );
+
+    return {
+      cutoff: response?.data?.meta?.cut_off,
+      nextPage: response?.data?.meta?.next_page_cursor,
+      transactions: consolidatedTransactions,
+    };
+  } catch (e) {
+    logger.error(new RainbowError('consolidatedTransactionsQueryFunction: '), {
+      message: e,
+    });
+    return { transactions: [] };
+  }
+}
+
+type ConsolidatedTransactionsResult = {
+  cutoff?: number;
+  nextPage?: string;
+  transactions: RainbowTransaction[];
+};
+/**
+ * 
+ * should we? 
+ *   queryClient.invalidateQueries({
+        queryKey: nftsQueryKey({ address: accountAddress }),
+      });
+ */
+async function parseConsolidatedTransactions(
+  message: TransactionsReceivedMessage,
+  currency: NativeCurrencyKey
+): Promise<RainbowTransaction[]> {
+  const data = message?.payload?.transactions || [];
+
+  return [];
```

Comment:
Not sure if you wanted to return an empty array here? Also tons of commented out code around here that could be cleaned up if not needed.